core/AutocorrectSuggestion.cc

```diff
@@ -26,6 +26,32 @@ bool hasSeen(const UnorderedSet<Loc> &seen, Loc loc) {
     return false;
 }
 
+string AutocorrectSuggestion::unsafeApplyToString(const std::string_view source) {
+    UnorderedSet<Loc> seen;
+    vector<AutocorrectSuggestion::Edit> edits = move(this->edits);
+
+    if (edits.size() > 1) {
+        auto file = edits.front().loc.file();
+        for (int i = 1; i < edits.size(); i++) {
+            ENFORCE(edits[i].loc.file() == file, "All edits in `unsafeApplyToString` need to apply to the same file");
+        }
+    }
+
+    auto compare = [](const AutocorrectSuggestion::Edit &left, const AutocorrectSuggestion::Edit &right) {
+        return left.loc.beginPos() > right.loc.beginPos();
+    };
+    fast_sort(edits, compare);
+
+    string replaced{source};
+    for (auto &edit : edits) {
+        auto start = edit.loc.beginPos();
+        auto end = edit.loc.endPos();
+        replaced = absl::StrCat(replaced.substr(0, start), edit.replacement, replaced.substr(end, -1));
+    }
+
+    return replaced;
```

Comment:
The documentation for this function in the header says that it applies a single edit, but the function as written here applies multiple edits.  The loop as written here relies on subsequent edits having locs in the mutated string rather than the original string, which seems like a subtle thing to get right.

I think this function should apply only a single edit and throw in the `0` or `> 1` edit cases, especially as this is really only for testing.  (And, to that end, should this be called `applySingleEditForTesting` or somesuch?)

Reply:
I think that makes sense. I can make that change.

Reply:
(FWIW, the documentation says it applies a single `AutocorrectSuggestion`, which is true: a single `AutocorrectSuggestion`, however, might have multiple edits.) This is in contrast to `AutocorrectSuggestion::apply`, which applies a vector of `AutocorrectSuggestion`s.)

Reply:
> (FWIW, the documentation says it applies a single AutocorrectSuggestion, which is true: a single AutocorrectSuggestion, however, might have multiple edits. This is in contrast to AutocorrectSuggestion::apply, which applies a vector of AutocorrectSuggestions.)

Doh, reading comprehension fail on my part.